### PR TITLE
Tasksception

### PR DESCRIPTION
**Pull request by jbachhardie**
Tasks can now run other tasks
https://trello.com/c/0etHCS0I/327-usher-run-tasks-from-other-tasks

### CODE REVIEW

File: src/run.js [hunk 1]
```diff
@@ -9,8 +9,14 @@ let TaskRunner = require('./modules/task-runner');
 function getTaskConfig(taskName, taskVars, opts) {
   const usherFile = opts.filepath || '.usher.yml';
   const config = parser.safeLoad(fs.readFileSync(usherFile, 'utf8'));
-  const splitVars = _.map(taskVars, a => a.split('='));
-  const parsedVars = _.fromPairs(splitVars);
+  let parsedVars
```
Comment (sashman) [inline src/run.js]: Needs a `;`

**Comment by sashman** [comment]
Would probably be a good idea to update the README, it's hard to tell how to use this w/o looking at the tests

File: test/test.usher.yml [hunk 2]
```diff
@@ -31,8 +31,14 @@ tasks:
...
   retry:
     - cmd: echo test_retry
       retry:
         attempts: 2
         delay: 0.1
+
+  run_task:
+    - task: build
+      vars:
+        version: task
```
Comment (sashman) [inline test/test.usher.yml:44]: Can you add another test with multiple tasks, and or command/task combinations

File: src/run.js [hunk 3]
```diff
@@ -9,8 +9,14 @@ let TaskRunner = require('./modules/task-runner');
...
+  if(Array.isArray(taskVars)) {
+    const splitVars = _.map(taskVars, a => a.split('='));
+    parsedVars = _.fromPairs(splitVars);
+  }
+  else {
+    parsedVars = taskVars
```
Comment (Angry-Potato) [inline src/run.js]: needs a ;

File: src/modules/task-runner.js [hunk 4]
```diff
@@ -30,8 +34,12 @@ class TaskRunner {
       Environment variables: ${env}`);
   }
 
+  runTask(command) {
+    logger.info(`Executing task : ${command.task} with vars ${JSON.stringify(command.vars)}`)
```
Comment (Angry-Potato) [inline src/modules/task-runner.js]: needs a ;, dirty javascript

**Comment by sashman** [comment]
Sounds like whoever is picking up the next usher change, will need to have the jscs linter installed. [Here](https://github.com/findmypast/curtain-twitcher/blob/master/.jscs.json) is the config that is used in other projects.

**Comment by sashman** [comment]
:+1:

**Pull request merged**